src/formatDate.js

```diff
@@ -7,8 +7,61 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newSeparator = toFormat[toFormat.length - 1];
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+
+  const dateParts = date.split(oldSeparator);
+
+  const yearIndexFrom =
+    fromFormat.indexOf('YYYY') !== -1
+      ? fromFormat.indexOf('YYYY')
+      : fromFormat.indexOf('YY');
+  const yearIndexTo =
+    toFormat.indexOf('YYYY') !== -1
+      ? toFormat.indexOf('YYYY')
+      : toFormat.indexOf('YY');
+  const monthIndexFrom = fromFormat.indexOf('MM');
+  const dayIndexFrom = fromFormat.indexOf('DD');
+
+  let year = dateParts[yearIndexFrom];
+  const month = dateParts[monthIndexFrom];
+  const day = dateParts[dayIndexFrom];
+
+  if (fromFormat[yearIndexFrom] === 'YY' && toFormat[yearIndexTo] === 'YYYY') {
+    if (year < 30) {
+      year = `20${year}`;
+    } else {
+      year = `19${year}`;
+    }
+  }
+
+  if (fromFormat[yearIndexFrom] === 'YYYY' && toFormat[yearIndexTo] === 'YY') {
+    year = year.slice(-2);
+  }
+
+  let newDate = '';
+
+  for (const part of toFormat.slice(0, -1)) {
+    switch (part) {
+      case 'YYYY':
+        newDate += year;
+        break;
+      case 'YY':
+        newDate += year;
+        break;
+      case 'MM':
+        newDate += month;
+        break;
+      case 'DD':
+        newDate += day;
+        break;
+    }
+    newDate += newSeparator;
+  }
+
+  return newDate.slice(0, -1);
```

Comment:
The function lacks error handling for cases where the input date does not match the expected `fromFormat`. It would be prudent to add validation for the input date format to ensure it aligns with the `fromFormat` array, as well as to check the validity of the date itself. Additionally, the function does not account for edge cases where the input year is '00'. The current logic would convert '00' to '2000', which may not always be the desired outcome. Consider adding a check or a parameter to specify the century cutoff year for better handling of such cases.